eth/fetcher/fetcher.go

```diff
@@ -253,14 +354,25 @@ func (f *Fetcher) loop() {
 				glog.V(logger.Debug).Infof("Peer %s: exceeded outstanding announces (%d)", notification.origin, hashLimit)
 				break
 			}
+			// If we have a valid block number, check that it's potentially useful
+			if notification.number > 0 {
+				if dist := int64(notification.number) - int64(f.chainHeight()); dist < -maxUncleDist || dist > maxQueueDist {
```

Comment:
Same here, how does this work with re-orgs larger than maxUncleDist or larger than maxQueueDist?